components/form/form.js

```diff
@@ -139,10 +139,14 @@ class Form extends FormMixin(LitElement) {
 		this._nestedForms.set(e.target, form);
 
 		const onFormDisconnect = () => {
-			form.removeEventListener('d2l-form-disconnect', onFormDisconnect);
-			this._nestedForms.delete(e.target);
+			this._nestedForms.forEach((value, key) => {
```

Comment:
Hmm... writing out a description here, I think this still might be slightly flawed. Kicking this back to draft while I investigate.

Reply:
Cool... I was just trying to follow this.

Is it true that the `const form = e.composedPath()[0];` above is the actual form element? And that `e.composedPath()[0]` of the `d2l-form-disconnected` event would be the same element?

It seems like the original code would have failed if there was more than one nested form too.

Reply:
Yeah exactly -- if there were more than 1 nested form inside a custom element, then the existing code would only know about the last one, since its `Map` is just from `e.target` (in this case the custom element) to form.

I'm going to rework this so that the `Map` contains an array of nested forms, and I think I can just have a single disconnect listener.

Reply:
Ok, sounds good. Yeah I was wondering if we really wanted multiple handlers wired up.

The alternative I was thinking before you mentioned that was something roughly along these lines...
```javascript
const onFormDisconnect = disconnectedEvent => {
    const disconnectedForm = disconnectedEvent.composedPath()[0];
    if (disconnectedForm === form) {
        ...
    }
}
```